libimage/filters_test.go

```diff
@@ -41,50 +42,62 @@ func TestFilterReference(t *testing.T) {
 	require.NoError(t, err)
 
 	for _, test := range []struct {
-		filter  string
+		filters []string
 		matches int
 	}{
-		{"image", 2},
-		{"*mage*", 2},
-		{"image:*", 2},
-		{"image:tag", 1},
-		{"image:another-tag", 1},
-		{"localhost/image", 1},
-		{"localhost/image:tag", 1},
-		{"library/image", 1},
-		{"docker.io/library/image*", 1},
-		{"docker.io/library/image:*", 1},
-		{"docker.io/library/image:another-tag", 1},
-		{"localhost/*", 2},
-		{"localhost/image:*tag", 1},
-		{"localhost/*mage:*ag", 2},
-		{"quay.io/libpod/busybox", 1},
-		{"quay.io/libpod/alpine", 1},
-		{"quay.io/libpod", 0},
-		{"quay.io/libpod/*", 2},
-		{"busybox", 1},
-		{"alpine", 1},
-		{"alpine@" + alpine.Digest().String(), 1},
-		{"alpine:latest@" + alpine.Digest().String(), 1},
-		{"quay.io/libpod/alpine@" + alpine.Digest().String(), 1},
-		{"quay.io/libpod/alpine:latest@" + alpine.Digest().String(), 1},
+		{[]string{"image"}, 2},
+		{[]string{"*mage*"}, 2},
+		{[]string{"image:*"}, 2},
+		{[]string{"image:tag"}, 1},
+		{[]string{"image:another-tag"}, 1},
+		{[]string{"localhost/image"}, 1},
+		{[]string{"localhost/image:tag"}, 1},
+		{[]string{"library/image"}, 1},
+		{[]string{"docker.io/library/image*"}, 1},
+		{[]string{"docker.io/library/image:*"}, 1},
+		{[]string{"docker.io/library/image:another-tag"}, 1},
+		{[]string{"localhost/*"}, 2},
+		{[]string{"localhost/image:*tag"}, 1},
+		{[]string{"localhost/*mage:*ag"}, 2},
+		{[]string{"quay.io/libpod/busybox"}, 1},
+		{[]string{"quay.io/libpod/alpine"}, 1},
+		{[]string{"quay.io/libpod"}, 0},
+		{[]string{"quay.io/libpod/*"}, 2},
+		{[]string{"busybox"}, 1},
+		{[]string{"alpine"}, 1},
+		{[]string{"alpine@" + alpine.Digest().String()}, 1},
+		{[]string{"alpine:latest@" + alpine.Digest().String()}, 1},
+		{[]string{"quay.io/libpod/alpine@" + alpine.Digest().String()}, 1},
+		{[]string{"quay.io/libpod/alpine:latest@" + alpine.Digest().String()}, 1},
+		// Make sure negate works as expected
+		{[]string{"!alpine"}, 1},
+		{[]string{"!alpine", "!busybox"}, 0},
+		{[]string{"!alpine", "busybox"}, 1},
```

Comment:
Perhaps also a test for two “wanted” patterns?

Reply:
Fixed